### ISSUE TITLE
cargo-c can't find pkg-config and fails in some custom build scripts

### ISSUE THREAD

**Issue by LigH-de** [m-ab-s/media-autobuild_suite#2207]
Excerpt of **ab-suite.rust.install.log**
```
  cargo-warning=Could not run `"G:/MABS/msys64/mingw32/bin/pkgconf --keep-system-libs --keep-system-cflags --static" "--libs" "--cflags" "zlib"`
  The pkg-config command could not be found.

  Most likely, you need to install a pkg-config package for your OS.

  If you've already installed it, ensure the pkg-config command is one of the
  directories in the PATH environment variable.
```

[logs.zip](https://0x0.st/oMQf.zip)

**Comment by LigH-de** [comment]
I believe to remember that the suite uses a custom call for pkg-config. I guess cargo tries to execute `pkgconf --keep-system-libs --keep-system-cflags --static` as the file name of an executable. I know there is a Windows function which assumes one string to be the name of the executable, and a second string to be the parameter list.

**Comment by 1480c1** [comment]
https://github.com/rust-lang/pkg-config-rs/blob/master/src/lib.rs#L474 found it

**Comment by kevev** [comment]
I am having this exact issue. Is there a workaround?

**Comment by LigH-de** [comment]
Not compiling anything related to rust until it's fixed ... so that's not a great advice.

**Comment by 1480c1** [comment]
pushed a commit https://github.com/m-ab-s/media-autobuild_suite/commit/53d4faf93a4b3184c1b47765418fe38f63c310a6, please try with that

**Comment by LigH-de** [comment]
Sorry...

```
cargo-warning=Failed to run command `"G:/MABS/local32/bin/ab-pkg-config" "--libs" "--cflags" "zlib"`, because: %1 ist keine zulässige Win32-Anwendung. (os error 193)
```
(ab-pkg-config is not a valid Win32 executable)

😡 

[logs.zip](https://0x0.st/oMva.zip)

**Comment by 1480c1** [comment]
Hmm, so I guess the pkg_config crate really only wants .exe files. This make take a little bit more time then

**Comment by kevev** [comment]
> pushed a commit [53d4faf](https://github.com/m-ab-s/media-autobuild_suite/commit/53d4faf93a4b3184c1b47765418fe38f63c310a6), please try with that

This worked!

**Comment by Barough** [comment]
Have the same issue here also
----------------------

┌ cargo-c git  ........................................ [Recently updated]
├ Running rust.update...
├ Running rust.install...
Likely error (tail of the failed operation logfile):
  running: "ccache" "gcc" "-O0" "-ffunction-sections" "-fdata-sections" "-m32" "-mthreads" "-mtune=generic" "-O2" "-pipe" "-I" "src/zlib" "-DZ_SOLO" "-o" "D:\\MABS-ffmpeg\\build\\cargo-c-git\\target\\release\\build\\libz-sys-4d66c68cf692c559\\out\\lib\\src/zlib/compress.o" "-c" "src/zlib/compress.c"
  exit code: 1

  --- stderr


  error occurred: Command "ccache" "gcc" "-O0" "-ffunction-sections" "-fdata-sections" "-m32" "-mthreads" "-mtune=generic" "-O2" "-pipe" "-I" "src/zlib" "-DZ_SOLO" "-o" "D:\\MABS-ffmpeg\\build\\cargo-c-git\\target\\release\\build\\libz-sys-4d66c68cf692c559\\out\\lib\\src/zlib/adler32.o" "-c" "src/zlib/adler32.c" with args "gcc" did not execute successfully (status code exit code: 1).


error: failed to compile `cargo-c v0.9.10+cargo-0.62 (D:\MABS-ffmpeg\build\cargo-c-git)`, intermediate artifacts can be found at `D:\MABS-ffmpeg\build\cargo-c-git\target`
rust.install failed. Check D:/MABS-ffmpeg/build/cargo-c-git/ab-suite.rust.install.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

All relevant logs have been anonymously uploaded to https://0x0.st/oMDD.zip

**Comment by 1480c1** [comment]
After trying to investigate why the builds were passing just fine with my clang builds but not with my gcc builds, all I could gather was that the gcc builds were trying to compile the vendored dependencies inside the `target` folder, but the clang builds were properly pulling from `/opt/cargo/registry....`. I can't tell if that's correct due to the amount of indirection used in rust to just run gcc, and it doesn't even show the stderr from gcc as far as I can tell. Comparing the two compile lines, the main difference (other than the compiler and compiler specific flags) is that the path uses `//` for gcc while clang uses `\\` for the path to the registry directory.

**Comment by Barough** [comment]
This issue started 5 days ago and cargo-c v0.9.10 was released 5 days ago. Just saying.......

**Comment by 1480c1** [comment]
The issue I was pointing out is related to dssim with lcms2-sys and mozjpeg-sys rather than cargo-c

**Comment by ultrasound1372** [comment]
Any news on this? Should I just disable dssim and mozjpeg to make my builds work again?

**Comment by 1480c1** [comment]
I believe https://github.com/m-ab-s/media-autobuild_suite/commit/87ece8ec16088504a341f812706aba944f13dcd8 should have addressed this for now

**Comment by Barough** [comment]
cargo-c compiles now

**Comment by LigH-de** [comment]
Yes, using clang for cargo-c, among gcc for other tools, works for me.
